extensions/grpc/runtime/src/main/java/io/quarkus/grpc/runtime/supports/RequestScopeHandlerInterceptor.java

```diff
@@ -0,0 +1,41 @@
+package io.quarkus.grpc.runtime.supports;
+
+import io.grpc.ForwardingServerCall;
+import io.grpc.Metadata;
+import io.grpc.ServerCall;
+import io.grpc.ServerCallHandler;
+import io.grpc.ServerInterceptor;
+import io.grpc.Status;
+import io.quarkus.arc.Arc;
+import io.quarkus.arc.ManagedContext;
+
+public class RequestScopeHandlerInterceptor implements ServerInterceptor {
+
+    private final ManagedContext reqContext;
+
+    public RequestScopeHandlerInterceptor() {
+        reqContext = Arc.container().requestContext();
+    }
+
+    @Override
+    public <ReqT, RespT> ServerCall.Listener<ReqT> interceptCall(ServerCall<ReqT, RespT> call,
+            Metadata headers,
+            ServerCallHandler<ReqT, RespT> next) {
+
+        boolean managed = !reqContext.isActive();
```

Comment:
why is this variable called `managed`?

Reply:
because I suck at naming. It just means that the request scope is "managed" by the interceptor (and not be enabled beforehand)